watcher.js

```diff
@@ -1,149 +0,0 @@
-                            workflowInstanceId: task.workflowInstanceId, 
-                            taskId: task.taskId, 
-                            reasonForIncompletion,
-                            status,
-                            ...taskUpdateData,
-                        })
-                } catch (e) {
-                    logger.error(e, 'Failed to update task for task type: %s', this.taskType)
-                }
-            }
-            callbackUpdater.complete = (_a) => callbackUpdater(_a, connector.TaskStatus.COMPLETED)
-            callbackUpdater.fail = (_a) => callbackUpdater(_a, connector.TaskStatus.FAILED)
-            callbackUpdater.failWithTerminalError = (_a) => callbackUpdater(_a, connector.TaskStatus.FAILED_WITH_TERMINAL_ERROR)
-            callbackUpdater.inprogress = (_a) => callbackUpdater({callbackAfterSeconds: ONE_YEAR_IN_SECONDS, ..._a}, connector.TaskStatus.IN_PROGRESS)
-            
-            callbackUpdater.log = (log) => this.log(task.taskId, log)
-            return callbackUpdater
-        }
-        
-        this.pollAndRepeat = async () => {
-            while (true) {
-                this.startTime = new Date()
-                try {
-                    if (this.isPolling) {
-                        const { baseURL, workerID } = this.options
-                        const freeRunnersCount = this.options.maxRunner - Object.keys(this.tasks).length
-                        if (freeRunnersCount > 0) {
-                            let resp = await connector.pollForTasks(baseURL, this.taskType, workerID, freeRunnersCount)
-                            if (this.pollingFailures === undefined || this.pollingFailures > 0) {
-                                logger.info('Initial polling started successfully for task type: %s', this.taskType)
-                                this.pollingFailures = 0
-                            }
-
-                            const tasks = resp.body || []
-                            await Promise.all(tasks.map(this.doCallbackForTask))
-                        }
-                    }
-                } catch (error) {
-                    this.pollingFailures = (this.pollingFailures || 0) + 1
-                    logger.error('Polling connection failed for task type: %s, error: %s, stack: %s', this.taskType, error.message, error.stack)
-                }
-
-                let pollInterval = this.options.pollingIntervals
-                if (this.options.pollFailureBackoffFactor && this.pollingFailures > 0) {
-                    pollInterval = pollInterval + (this.options.pollFailureBackoffFactor * pollInterval * this.pollingFailures)
-                    
-                    if (this.options.maxPollInterval && pollInterval > this.options.maxPollInterval) {
-                        pollInterval = this.options.maxPollInterval
-                    }
-                }
-
-                await sleep(pollInterval - (new Date().getTime() - this.startTime.getTime()))
```

Comment:
Conductor Java (and other clients AFAIK) usually just poll with the full interval so I did not re-implement this.